scalafix-interfaces/src/main/java/scalafix/interfaces/ScalafixArguments.java

```diff
@@ -205,4 +205,11 @@ ScalafixArguments withToolClasspath(
      */
     ScalafixError[] run();
 
+    /**
+     * Similar to {@link #run()}, but without any side effects on the source files. Via the returned {@link ScalafixEvaluation},
+     * for each file, diagnostics can be inspected, and patches can be previewed and applied.
+     * <p>
+     * Incompatible with {@link #withMainCallback} and {@link #withMode}.
```

Comment:
can this be tested? I don't see a check in the implementation

Reply:
Yes, I will add test for this.